gap/PGMatrix.gd

```diff
@@ -1,93 +1,203 @@
+#!   <A>symmetries</A>, given as words written in terms of the rotation generators `a`, `b`, `c` 
+#!   of the triangle group $\Delta$ or the reflection generators `x`, `y`, `z` of the proper triangle 
+#!   group $\Delta^+$, specified by the `TriangleGroup` object  <A>fulltg</A> and `ProperTriangleGroup`
+#!   object <A>tg</A>, respectively (see <Ref Sect="Section_TriangleGroups"/>). Each point-group
+#!   matrix is constructed through a sequence of matrix multiplication of the point-group matrices
+#!   in <A>pgMatElements</A>, i.e., the point-group matrices of the generators `a`, `b` and `c`
+#!   of $\Delta$. The sequences are specified by the corresponding words in <A>symmetries</A>.
+#!
 #!   The option `symNames` can be used to specify the names of the <A>symmetries</A>. If a 
 #!   single symmetry is provided in the argument <A>symmetries</A>, `symNames` takes a single
 #!   string, otherwise a list of strings with the same number of elements as in <A>symmetries</A>
-#!   must be provided. The names will be used to label the corresponding point-group matrices.
-#!   The option `sparse`sparse, which takes a boolean, can be used to generate a sparse 
-#!   representation of the adjacency matrix. If sparse is true the adjacency matrix is of the form
-#!   `[ [ [ rowIdx, colIdx ], entry ], ... ]`, where `entry` is the corresponding matrix entry at
-#!   position `rowIdx` and `colIdx`, which represent indices of the matrix rows and columns, 
-#!   respectively. The default is false.
-#! @Arguments symmetries, fulltgFp, cell, quotient
-#! @Returns list of point-group matrices
+#!   must be provided. The names will be used to label the corresponding point-group matrices. 
+#!   If `symNames` is not specified, the words in <A>symmetries</A> will used as labels instead.
+#!   
+#!   If  the option `sparse` was used in the construction of <A>pgMatElements</A>, the point-group 
+#!   matrices will be sparsly represented as well and will be of the form `[ [ [ rowIdx, colIdx ], entry ], ... ]`, 
+#!   where `entry` is the corresponding matrix element at position `rowIdx` and `colIdx`, which represent
+#!   indices of the matrix rows and columns, respectively. 
+#! @Arguments symmetries, fulltg, tg, pgMatElements
+#! @Returns list of point-group matrices as a `PGMatrix` object.
 DeclareGlobalFunction( "PGMatrix" );
 
-#! 
-#! The generated `PGMatrix` is printed in the form
-#! @BeginLog
-#! [
-#!   signature,
-#!   quotient,
-#!   [ [symName1, PGMatrix1], [symName2, PGMatrix2], ... ]
-#! ]
-#! @EndLog
+
+#! @Description
+#!   returns the point-group matrices of the (full) triangle group generators `a`, `b` and `c` as a
+#!   `PGMatrixElements` object.
+#! @Arguments pgMat
+#! @Label for PGMatrix
+DeclareOperation( "GetPGMatrixElements", [ IsPGMatrixObj ] );
+
+
+#! @Description
+#!   returns a list of tuples. Each tuple is of the form `[symmetryName, PGMatrix]`, where `symmetryName` 
```

Comment:
If we change the name of the object from `PGMatrix` to something else, than this is fine, but at the moment this is misleading since one could think that `PGMatrix` in the tuple is actually an object of that form rather than a matrix, maybe just call it `matrix` instead?